client/js/lounge.js

```diff
@@ -428,16 +490,20 @@ $(function() {
 		}
 
 		if (prevMsgTime.toDateString() !== msgTime.toDateString()) {
-			prevMsg.append(render("date-marker", {msgDate: msgTime}));
+			var parent = prevMsg.parent();
+			if (parent.hasClass("condensed")) {
+				prevMsg = parent;
+			}
+			prevMsg.after(render("date-marker", {msgDate: msgTime}));
```

Comment:
Yeah, because append adds **into** something, but we don't want the date-marker to be inside the messae, we want it to be after it.